packages/compass-saved-aggregations-queries/src/components/select-connection-modal.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback } from 'react';
+import {
+  FormModal,
+  Radio,
+  RadioGroup,
+  css,
+  spacing,
+} from '@mongodb-js/compass-components';
+import {
+  type MapDispatchToProps,
+  type MapStateToProps,
+  connect,
+} from 'react-redux';
+import { type RootState } from '../stores';
+import {
+  closeModal,
+  connectionSelected,
+  openSelectedItem,
+} from '../stores/open-item';
+
+const modalContent = css({
+  display: 'grid',
+  gridAutoColumns: '1fr',
+  rowGap: spacing[4],
+  columnGap: spacing[3],
+  gridTemplateAreas: `
+    'description'
+    'connection'
+  `,
+});
+
+const description = css({
+  gridArea: 'description',
+});
+
+const connection = css({
+  gridArea: 'connection',
+});
+
+type SelectConnectionModalProps = {
+  namespace: string;
+  itemType: string;
+  itemName: string;
+  isModalOpen: boolean;
+  isSubmitDisabled: boolean;
+  connections: { id: string; name: string }[];
+  selectedConnection: string | null;
+  selectedDatabase: string | null;
+  selectedCollection: string | null;
+  onSelectConnection(
+    connectionId: string,
+    database: string | null,
+    collection: string | null
+  ): void;
+  onSubmit(): void;
+  onClose(): void;
+};
+
+const SelectConnectionModal: React.FunctionComponent<
+  SelectConnectionModalProps
+> = ({
+  namespace,
+  itemType,
+  itemName,
+  isModalOpen,
+  isSubmitDisabled,
+  connections,
+  selectedConnection,
+  selectedDatabase,
+  selectedCollection,
+  onSelectConnection,
+  onSubmit,
+  onClose,
+}) => {
+  const handleConnectionSelect: React.ChangeEventHandler<HTMLInputElement> =
+    useCallback(
+      (event) => {
+        onSelectConnection(
+          event.target.value,
+          selectedDatabase,
+          selectedCollection
+        );
```

Comment:
this part feels off - we get the selectedDatabase and selectedConnection from redux and you cannot change them here, why are we sending it back then? then onSelectConnection could just take the connectionId, which would also make it less confusing.

Reply:
Yea reading it back again definitely felt wrong. I was overloading the `connectionSelected` action while trying to cover the two events happening in app:
1) connection getting selected and database and collection dropdown getting reset (in select-connection-and-namespace-modal)
2) connection getting selected for pre selected database and collection (in select-connection-modal)

I have created another action in [e96fca4](https://github.com/mongodb-js/compass/pull/5841/commits/e96fca4c1a2da10a41469ca139464beb518ba9d9), to exactly address the second event. That commit should address this. Let me know what you think.

Reply:
Ah, so it is to not reset the namespace, gotcha. Looks much clearer now with the separate function, thank you!